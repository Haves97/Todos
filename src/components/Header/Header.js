import React from 'react';
import cl from './Header.module.css'

const Header = () => {
    return (
        <header >
            <h1 className={cl.title}>todos</h1>
        </header>
    );
};

export default Header;