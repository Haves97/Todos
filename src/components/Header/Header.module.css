.title {
    margin: 0;
    text-align: center;
    font-size: 80px;
    font-weight: 100;
    color: rgba(175, 47, 47, 0.15);
}