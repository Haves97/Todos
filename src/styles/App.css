body {
    font-family: 'Roboto', sans-serif;
    font-size: 30px;
    font-weight: 400;
    background-color: #dfe6e9;
}

.App {
    max-width: 500px;
    margin: 0 auto;
    
}