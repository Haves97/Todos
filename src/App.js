import { useState } from "react";
import Header from "./components/Header/Header";
import TodoForm from "./components/TodoForm/TodoForm";
import TodoList from "./components/TodoList/TodoList";
import useAddTodo from "./hooks/useAddTodo";
import useSortedTodos from "./hooks/useSortedTodos";
import useCheckboxState from "./hooks/useCheckboxState"
import './styles/App.css'

function App() {
  const { todos, addTodo } = useAddTodo([]);
  const { checked, onChange } = useCheckboxState({});
  const [sort, setSort] = useState('all');
  const sortTodos = useSortedTodos(sort, todos, checked);
  return (
    <div className="App">
      <Header />
      <TodoForm addTodo={addTodo} />
      <TodoList
        todos={sortTodos}
        sort={sort}
        setSort={setSort}
        checked={checked}
        onChange={onChange}
      />
    </div>
  );
}

export default App;
