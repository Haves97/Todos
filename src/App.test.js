import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';
import Header from './components/Header/Header'


// describe("App", () => {
//   it("renders App component", () => {
//   });
// });

describe("App", () => {
  it("renders App component", () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});


describe("Header", () => {
  it("Get text 'todos'", () => {
    const { getByText } = render(<Header />);
    const linkElement = getByText(/todos/i);
    expect(linkElement).toBeInTheDocument();
  });
});

describe("Checkbox", () => {
  it("Checkbox click", () => {
    const { container } = render(<Che)
    // userEvent
  });
});