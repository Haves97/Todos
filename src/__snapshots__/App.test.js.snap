// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header>
      <h1
        class="title"
      >
        todos
      </h1>
    </header>
    <form
      class="form"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-b9ayx-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-textarea"
            placeholder="Add todo"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
    <ul
      class="MuiList-root list css-1mk9mw3-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-divider css-1g8g675-MuiListItem-root"
      >
        <div
          class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
          role="group"
        >
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1yx66g7-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="all"
          >
            all
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1yx66g7-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="active"
          >
            active
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1yx66g7-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="completed"
          >
            completed
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
